src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,50 @@
 package core.basesyntax;
 
-class Pair {
+public class Pair<T1, T2> {
+    private T1 t1;
+    private T2 t2;
 
+    private Pair(T1 t1, T2 t2) {
+        this.t1 = t1;
+        this.t2 = t2;
+    }
+
+    public static <T1, T2> Pair<T1, T2> of(T1 t1, T2 t2) {
+        return new Pair<>(t1, t2);
+    }
+
+    public T1 getFirst() {
+        return t1;
+    }
+
+    public T2 getSecond() {
+        return t2;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        Pair<?, ?> pair = (Pair<?, ?>) o;
```

Comment:
what's the purpose of using wildcard here?

Reply:
intellij idea told me it would be nice to avoid unchecked cast. not sure I understand how to use wildcard right, so, fixed

Reply:
you will learn about wildcard soon 👍